src/core/sync/mutex.d

```diff
@@ -377,6 +377,7 @@ unittest
     // For example, Bionic doesn't appear to do so, so this test is
     // not run on Android.
     version (CRuntime_Bionic) {} else
+    version (DragonFlyBSD) {} else               // FIXME : Will have to write a test program to see what is going on here.
```

Comment:
// FIXME : 
Will have to write a little c-program to check if DragonFlyBSD pthread does allow trylock after the mutex has been destroyed and does return 0 in this case.

Reply:
Have not found a good solution for this. But trying to lock after destroying the lock does not really make sense anyway. Reading up on some of the documentation regarding destroying a lock, it looks like it is underfined behaviour to try and lock the mutex after destruction. Even though it seems to "work" in some implementations.

Reply:
Leaving it like this for now. 
Would require changing pthread implementation on dragonfly, or adding a check to "class Mutex" to verify if it has been initialized/constructed. Both don't really seem reasonable.